site/content/blog/2019-01-20-submitting-great-prs.md

```diff
@@ -0,0 +1,76 @@
+
+Now, not everything deserves to be fixed in every pull request. Fixing a one-line bug isn't usually the time to fix an unrelated variable naming issue that touches multiple files. This leads to my second point:
+
+### 2) Give a great description.
+
+The description on a pull request is your chance to prove to a reviewer that you have the credentials to fix this problem. How? State the problem in your own words:
+
+> This pull request adds new functionality to expose ...
+
+Tell the reviewer why we need to do this:
+
+> With this new functionality, we can fix a longstanding customer issue that ...
+
+You don't need to outline everything your code is doing -- that's why the code is there -- but this is also your opportunity to point the reviewer in the right direction:
+
+> The trickiest part of this pull request was the `SqlStore` changes, because ...
+
+and also:
+
+> I really wanted to fix the variable names in this file, but it's out of scope for this pull request.
+
+Of course, sometimes you have to fix something else before you can get to the issue at hand. Maybe you decide to fix that variable name problem because it clarifies the feature you're adding. On to my third point:
+
+### 3) Clean up your history.
+
+Everyone has their own development style. Mine usually consists of coding a small part of the overall problem, typing:
+
+```sh
+git commit -m wip
+```
+
+and repeating this process often. Commit early and commit often! It's just easier to undo changes this way while you're actively developing.
+
+But a reviewer doesn't need to see that commit history. As a reviewer myself, I'd want to see one of two things. Either:
+
+1. A single squashed commit with all of your changes.
+
+2. Multiple commits breaking up your changes into logical groupings.
+
+Put that variable name fix in its own commit, commit your feature change on top of it, and add a note to your PR's description. Your reviewers can then use the commits tab in GitHub to review those changes independently.
+
+Maybe you're the kind of developer that writes a pristine commit history the first time, but most of us rely on `git rebase`. If you aren't familiar with rebasing, it might seem terrifying or magical, but it doesn't need to be: play with it, master it, and use it often. If you need help getting started here, check out Atlassian's excellent [tutorial on git rebase](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).
+
+Keep in mind though that rebasing is something to do before you submit your pull request, not after, and this leads to my last point:
+
+### 4) Avoid Force Pushing
+
+Good git etiquette is avoid force pushing any branch someone else is actively using. A pull request is a branch just like any other, so once you've started that pull request, everything that you push upstream should just be a regular or merge commit. In particular, when merge conflicts arise with `master`, just:
```

Comment:
```suggestion
Good git etiquette is to avoid force pushing any branch someone else is actively using. A pull request is a branch just like any other, so once you've started that pull request, everything that you push upstream should just be a regular or merge commit. In particular, when merge conflicts arise with `master`, just:
```